### PR TITLE
Add tutorial on migrating the Light class from gazebo classic

### PR DESCRIPTION
**Pull request by iche033**
# 🎉 New feature

Part of https://github.com/gazebosim/gz-sim/issues/325

## Summary

Add migration tutorial for Gazebo classic's Light class.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: tutorials/migration_light_api.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+In Gazebo, the light APIs has been consolidated into a single Light class with
+some of generic functions available through other utility / core classes.
+You'll find the APIs below on the following headers:
+
+* [ignition/gazebo/Light.hh](https://gazebosim.org/api/gazebo/6/Light_8hh.html)
+* [ignition/gazebo/Util.hh](https://gazebosim.org/api/gazebo/6/Util_8hh.html)
+* [ignition/gazebo/SdfEntityCreator.hh](https://gazebosim.org/api/gazebo/6/SdfEntityCreator_8hh.html)
+* [ignition/gazebo/EntityComponentManager.hh](https://gazebosim.org/api/gazebo/6/classignition_1_1gazebo_1_1EntityComponentManager.html)
+
+It's worth remembering that most of this functionality can be performed using
```
Comment (j-rivero) [inline tutorials/migration_light_api.md:37]: Can we elaborate a bit on this? How should someone use the `EntityComponentManager` to access to the light functionality?

File: tutorials/migration_light_api.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+the
+[EntityComponentManager](https://gazebosim.org/api/gazebo/6/classignition_1_1gazebo_1_1EntityComponentManager.html)
+directly. The functions presented here exist for convenience and readability.
```
Comment (j-rivero) [inline tutorials/migration_light_api.md:40]: What `here` means here :roll_eyes: ?

File: tutorials/migration_light_api.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+CastShadows | `ignition::gazebo::Light::CastShadows`
+Clone | TODO
+DiffuseColor | `ignition::gazebo::Light::DiffuseColor`
+Direction | `ignition::gazebo::Light::Direction`
+FillMsg | TODO
+Id | `ignition::gazebo::Light::Entity`
+LightType | `ignition::gazebo::Light::Type`
+Name | `ignition::gazebo::Light::Name`
+Position | `ignition::gazebo::Light::Pose`
+Rotation | `ignition::gazebo::Light::Pose``
```
Comment (j-rivero) [inline tutorials/migration_light_api.md:64]:
```suggestion
Rotation | `ignition::gazebo::Light::Pose`
```

File: tutorials/migration_light_api.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+SetAttenuation | use `ignition::gazebo::Light::SetAttenuation\*`
```
Comment (j-rivero) [inline tutorials/migration_light_api.md:65]:
```suggestion
SetAttenuation | use `ignition::gazebo::Light::SetAttenuation`
```

File: tutorials/migration_light_api.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+SetLightType | TODO
+SetName | TODO
+SetPosition | TODO
+SetRange | `ignition::gazebo::Light::SetAttenuationRange`
+SetRotation | TODO
+SetSelected |  Selection is client-specific, not porting
+SetSpecularColor | `ignition::gazebo::Light::SetSpecularColor`
+SetSpotFalloff | `ignition::gazebo::Light::SetSpotFalloff`
+SetSpotInnerAngle | ignition::gazebo::Light::SetSpotInnerAngle`
+SetSpotOuterAngle | ignition::gazebo::Light::SetSpotOuterAngle`
```
Comment (j-rivero) [inline tutorials/migration_light_api.md:78]:
```suggestion
SetSpotOuterAngle | `ignition::gazebo::Light::SetSpotOuterAngle`
```

File: tutorials/migration_light_api.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+---
+
+Classic | Gazebo
+-- | --
+AddChild | Not supported
+RemoveChild | Not supported
+RemoveChildren | Not supported
+SetParent | TODO
+SetWorld | TODO
+
```
Comment (j-rivero) [inline tutorials/migration_light_api.md:128]: I assume that is currently no possible to use a write-like method in Gazebo, is this correct? If so, we can mention it.

File: tutorials/migration_light_api.md
Comment (j-rivero) [inline tutorials/migration_light_api.md:1]: The acronym `TODO` is widely used in this document, I assume that means "It has not been implemented yet". Maybe worth mentioning.

**Review by j-rivero — CHANGES_REQUESTED** [review]
Some fixes and comments.

File: tutorials/migration_light_api.md (hunk 2 above)
Comment (iche033) [reply tutorials/migration_light_api.md]: clarified what `here` means in [4320596](https://github.com/gazebosim/gz-sim/pull/1931/commits/4320596ba96f6ab062d8cc2d52f12a561d3eefe5)

File: tutorials/migration_light_api.md (hunk 3 above)
Comment (iche033) [reply tutorials/migration_light_api.md]: fixed [4320596](https://github.com/gazebosim/gz-sim/pull/1931/commits/4320596ba96f6ab062d8cc2d52f12a561d3eefe5)

File: tutorials/migration_light_api.md (hunk 4 above)
Comment (iche033) [reply tutorials/migration_light_api.md]: ah the `*` is supposed to be there (for wildcard match). Removed `\`. [4320596](https://github.com/gazebosim/gz-sim/pull/1931/commits/4320596ba96f6ab062d8cc2d52f12a561d3eefe5)

File: tutorials/migration_light_api.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+SetDirection | `ignition::gazebo::Light::SetDirection`
+SetLightType | TODO
+SetName | TODO
+SetPosition | TODO
+SetRange | `ignition::gazebo::Light::SetAttenuationRange`
+SetRotation | TODO
+SetSelected |  Selection is client-specific, not porting
+SetSpecularColor | `ignition::gazebo::Light::SetSpecularColor`
+SetSpotFalloff | `ignition::gazebo::Light::SetSpotFalloff`
+SetSpotInnerAngle | ignition::gazebo::Light::SetSpotInnerAngle`
```
[the remark this answers is not in the document]
Comment (iche033) [reply tutorials/migration_light_api.md]: fixed. [4320596](https://github.com/gazebosim/gz-sim/pull/1931/commits/4320596ba96f6ab062d8cc2d52f12a561d3eefe5)

File: tutorials/migration_light_api.md (hunk 5 above)
Comment (iche033) [reply tutorials/migration_light_api.md]: fixed [4320596](https://github.com/gazebosim/gz-sim/pull/1931/commits/4320596ba96f6ab062d8cc2d52f12a561d3eefe5)

File: tutorials/migration_light_api.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+SetSpotOuterAngle | ignition::gazebo::Light::SetSpotOuterAngle`
+SetVisible | TODO
+ShowVisual | TODO
+SpecularColor | ignition::gazebo::Light::SetSpecularColor`
```
[the remark this answers is not in the document]
Comment (iche033) [reply tutorials/migration_light_api.md]: fixed [4320596](https://github.com/gazebosim/gz-sim/pull/1931/commits/4320596ba96f6ab062d8cc2d52f12a561d3eefe5)

File: tutorials/migration_light_api.md
Comment (iche033) [reply tutorials/migration_light_api.md:1]: elaboarated on what `TODO` means. [20d0011](https://github.com/gazebosim/gz-sim/pull/1931/commits/20d001183d25dd91cec3d595586120578b11e6ee)

**Review by j-rivero — APPROVED** [review]
Thanks @iche033 . Looks great!

**Pull request merged**